mmv1/third_party/terraform/provider/provider.go.erb

```diff
@@ -703,6 +710,42 @@ func ProviderConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Pr
 			"GOOGLE_OAUTH_ACCESS_TOKEN",
 		})
 	}
+	
+	// set universe_domain based on the service account key file.
+	if config.Credentials != "" {
+		contents, _, err := verify.PathOrContents(config.Credentials)
+		if err != nil {
+			fmt.Errorf("error loading service account credentials: %s", err)
+		}
+		var content map[string]any
+
+		if err := json.Unmarshal([]byte(contents), &content); err != nil {
+			return nil, diag.FromErr(err)
+		}
+
+		if content["universe_domain"] != nil {
+			config.UniverseDomain = content["universe_domain"].(string)
+		}
+	}
+	
+	// Check if the user provided a value from the universe_domain field
+	if v, ok := d.GetOk("universe_domain"); ok {
+		if config.UniverseDomain == "" {
+			config.UniverseDomain = v.(string)
+		} else if  v.(string) != "googleapis.com" {
```

Comment:
What if they both match?

Reply:
The current rule follows strictly what's in the prd. For now if they both match an error will be returned.

Reply:
Ah, the PRD isn't self-consistent. What's written is "If defined explicitly and a different universe_domain is already present in the credential file, an error should be returned by the provider"

Reply:
Ok, that's a very reasonable statement.